### ISSUE TITLE
Allow form modification during `request-confirm`

### ISSUE THREAD

**Issue by sander-beenen** [octobercms/ajax#40]
In one of our projects we are using the promise of `ajax:confirm-message`, fired by a `data-request-confirm` attribute, to modify values in the form that is being confirmed. However, since the OctoberCMS 3.4 update, this is not working anymore. We are unsure with what version this has been broken. 

We need to use this event (and not another one), because we want to show a modal where the user gives their reason for making a change. We can do this by preventing the default behaviour, showing a modal with input value and after confirmation in the modal (and entering the input) resolving the promise. An example is provided at the end.

The reason for this is that the form is being serialized [here](https://github.com/octobercms/ajax/blob/main/src/request/request.js#L50). But the confirmation is executed [here](https://github.com/octobercms/ajax/blob/main/src/request/request.js#L74), a couple of lines below. This means that during the confirmation there is no possibility to modify the form data.

Is it possible to change this?

Some example code to show what used to work:

```html
<form data-request="onDemo" data-request-confirm="Please confirm">
    <input type="hidden" id="demo-field" name="demo" />
    <button type="submit">Submit</button>
</form>
```

```js
$(window).on('ajax:confirm-message', function (event) {
    event.preventDefault();

    let promise = event.detail.promise;

    // this would be a modal which the user can submit in our real world scenario. 
    setTimeout(function () {
        $('#demo-field').val('some-value');

        promise.resolve();
    }, 5000);

    return true;
});
```

**Comment by daftspunk** [comment]
Hi @sander-beenen

Thanks for finding this. It is a slight inaccuracy from porting the jQuery version to the vanilla JS version. 

You can patch this by locally including the latest distribution in your theme and referencing it instead of the core version: https://github.com/octobercms/ajax/tree/main/dist

This fix will be included in v3.5.8

**Comment by sander-beenen** [comment]
Thanks for the quick fix and response @daftspunk!
